### PR TITLE
Dan Northcott

### PR DESCRIPTION
**Pull request by Dan2024**

### CODE REVIEW

File: domain-model.md [hunk 1]
```diff
@@ -0,0 +1,12 @@
...
+|               |                                                            |                  |                                                    |                                            |
+| cohortManager | cohortList @array                                          | searchByCohort() | cohort @obj (inst.of.cohort)                       | newCohortList @array                       |
+|               |                                                            | removeCohort()   | cohort @obj (inst.of.cohort)                       | newCohortList @array                       |
+|               |                                                            | createCohort()   | name @string                                       | cohort (inst.of.cohort)                    |
+|               |                                                            |                  |                                                    |                                            |
+| student       | studentID, firstName, lastName, gitUsername, email @string |                  |                                                    |                                            |
```
Comment (Hoolio333) [inline domain-model.md:12]: Impressive you made your domain model look so clear & neat inside VS. Looks great.

File: spec/cohortmanager.spec.js [hunk 2]
```diff
@@ -1,29 +1,109 @@
...
+    // execute
+    cohortManager.addNewCohort(cohort1);
+    cohortManager.addStudentToCohort(student1, "Cohort 1");
+    cohortManager.addStudentToCohort(student2, "Cohort 1");
+
+    // verify
+    const result = cohortManager.removeStudentFromCohort(student2, "Cohort 1");
+
+    expect(result.studentList).toEqual([student1]);
+  });
```
Comment (Hoolio333) [inline spec/cohortmanager.spec.js]: Really like this test. Clear from each stage.

File: src/cohortmanager.js [hunk 3]
```diff
@@ -73,6 +73,25 @@ class CohortManager {
 
     return false
   }
+
+  searchByStudentFirstAndLastName(firstName, lastName) {
+    const firstNameArray = []
+    // let lastNameArray = []
+
+    for (let i = 0; i < this.cohortList.length; i++) {
+      const firstNameArrayFromEachCohort =
```
Comment (Hoolio333) [inline src/cohortmanager.js]: Longest variable I've seen yet - but it's clear! :D

File: src/cohort.js [hunk 4]
```diff
@@ -0,0 +1,48 @@
...
+
+  hasStudent(studentEmail) {
+    return this.getStudent(studentEmail) !== undefined
+  }
+
+  getStudent(studentEmail) {
+    return this.studentList.find((student) => student.email === studentEmail)
+  }
+
+  filterByStudentFirstOrLastName(firstOrLastName) {
```
Comment (Tibor22) [inline src/cohort.js:38]: Good old filter, nice one mate

File: src/cohortmanager.js [hunk 5]
```diff
@@ -0,0 +1,97 @@
...
+    return `Err: student not found`
+  }
+
+  cohortExists(cohortName) {
+    return this.searchByCohortName(cohortName) !== undefined
+  }
+
+  studentExistsInOtherCohort(studentObj, cohortName) {
+    for (let i = 0; i < this.cohortList.length; i++) {
+      if (this.cohortList[i].name === cohortName) continue
```
Comment (Tibor22) [inline src/cohortmanager.js:70]: Nice you skipp over you own cohort with continue

**Pull request closed without merging**